core/src/main/java/com/linecorp/armeria/client/endpoint/AbstractDynamicEndpointGroupBuilder.java

```diff
@@ -25,11 +30,23 @@
 public abstract class AbstractDynamicEndpointGroupBuilder implements DynamicEndpointGroupSetters {
 
     private boolean allowEmptyEndpoints = true;
+    private long selectionTimeoutMillis;
 
     /**
      * Creates a new instance.
      */
-    protected AbstractDynamicEndpointGroupBuilder() {}
+    protected AbstractDynamicEndpointGroupBuilder() {
+        this(Flags.defaultConnectTimeoutMillis());
```

Comment:
Sorry for the late response.

> I think the default for DnsEndpointGroupBuilder will be connectTimeout instead of responseTimeout. I wanted to check if this is your intention.

Currently, the default select timeout for `DnsEndpointGroup` is the default connection timeout. 
https://github.com/line/armeria/blob/af45d0b2ae59c1a6c5a543257871b12dca92edfe/core/src/test/java/com/linecorp/armeria/client/endpoint/SelectionTimeoutTest.java#L118
DNS servers usually return a response quickly because they cache DNS records. Therefore, 3.2 seconds may be reasonable.

By the way, I checked the default timeout for a DNS query. It is 5 seconds that can be a more sensible default timeout for `DnsEndpointGroup/

> what do you think of just removing the empty constructor?

As this class is a public API, that could cause a breaking change.

Reply:
> As this class is a public API, that could cause a breaking change.

AbstractDynamicEndpointGroupBuilder is part of unstable APIs. Will remove the default constructor.

Reply:
It sounds reasonable to use the same timeout for `DnsEndpointGroup` and our DNS resolver.